geode-core/src/test/java/org/apache/geode/internal/cache/TXLastEventInTransactionUtilsTest.java

```diff
@@ -0,0 +1,192 @@
+
+  Cache cache;
+  InternalRegion region1;
+  InternalRegion region2;
+  InternalRegion region3;
+  InternalRegion region4;
+  InternalRegion region5;
+  InternalRegion region6;
+  InternalRegion region7;
+  InternalRegion region8;
+
+  GatewaySender sender1;
+  GatewaySender sender2;
+  GatewaySender sender3;
+  GatewaySender sender4;
+
+  @Before
+  public void setUp() {
+    sender1 = mock(GatewaySender.class);
+    when(sender1.isGroupTransactionEvents()).thenReturn(false);
+    sender2 = mock(GatewaySender.class);
+    when(sender2.isGroupTransactionEvents()).thenReturn(false);
+    sender3 = mock(GatewaySender.class);
+    when(sender3.isGroupTransactionEvents()).thenReturn(true);
+    sender4 = mock(GatewaySender.class);
+    when(sender4.isGroupTransactionEvents()).thenReturn(true);
+
+    cache = mock(Cache.class);
+    when(cache.getGatewaySender(SENDER_1)).thenReturn(sender1);
+    when(cache.getGatewaySender(SENDER_2)).thenReturn(sender2);
+    when(cache.getGatewaySender(SENDER_3)).thenReturn(sender3);
+    when(cache.getGatewaySender(SENDER_4)).thenReturn(sender4);
+    when(cache.getGatewaySender(SENDER_5)).thenReturn(null);
+
+    final String listSenderIdsForRegion1_2[] = {SENDER_1, SENDER_2};
+    final String listSenderIdsForRegion3_4[] = {SENDER_3, SENDER_4};
+    final String listSenderIdsForRegion5_6[] = {SENDER_1, SENDER_3};
+    final String listSenderIdsForRegion7[] = {SENDER_3};
+    final String listSenderIdsForRegion8[] = {SENDER_5};
+
+
+    Set senderIdsForRegion1_2 = new HashSet(Arrays.asList(listSenderIdsForRegion1_2));
+    Set senderIdsForRegion3_4 = new HashSet(Arrays.asList(listSenderIdsForRegion3_4));
+    Set senderIdsForRegion5_6 = new HashSet(Arrays.asList(listSenderIdsForRegion5_6));
+    Set senderIdsForRegion7 = new HashSet(Arrays.asList(listSenderIdsForRegion7));
+    Set senderIdsForRegion8 = new HashSet(Arrays.asList(listSenderIdsForRegion8));
+
+    region1 = mock(InternalRegion.class);
+    when(region1.getAllGatewaySenderIds()).thenReturn(senderIdsForRegion1_2);
+
+    region2 = mock(InternalRegion.class);
+    when(region2.getAllGatewaySenderIds()).thenReturn(senderIdsForRegion1_2);
+
+    region3 = mock(InternalRegion.class);
+    when(region3.getAllGatewaySenderIds()).thenReturn(senderIdsForRegion3_4);
+
+    region4 = mock(InternalRegion.class);
+    when(region4.getAllGatewaySenderIds()).thenReturn(senderIdsForRegion3_4);
+
+    region5 = mock(InternalRegion.class);
+    when(region5.getAllGatewaySenderIds()).thenReturn(senderIdsForRegion5_6);
+
+    region6 = mock(InternalRegion.class);
+    when(region6.getAllGatewaySenderIds()).thenReturn(senderIdsForRegion5_6);
+
+    region7 = mock(InternalRegion.class);
+    when(region7.getAllGatewaySenderIds()).thenReturn(senderIdsForRegion7);
+
+    region8 = mock(InternalRegion.class);
+    when(region8.getAllGatewaySenderIds()).thenReturn(senderIdsForRegion8);
+  }
+
+  @Test
+  public void noSenderGroupingTransactions() throws Exception {
```

Comment:
How about `getLastTransactionEventThrowsExceptionWhenSenderNotFound`?

Reply:
This test doesn't seem to be asserting that an exception is thrown anywhere, so that name would be confusing. Perhaps `getLastTransactionEventReturnsNullWhenGroupTransactionEventsIsFalseForAllSenders`?

Reply:
I might have added the wrong reply here.
The intended proposal was: `lastTransactionEventNotFoundWhenNoSenderGroupsTransactions`.
Anyway, I like your proposal so I will stick with it :-)